src/Microsoft.DotNet.Wpf/src/WindowsBase/System/Windows/DependencyProperty.cs

```diff
@@ -979,19 +977,17 @@ internal static DependencyProperty FromName(string name, Type ownerType)
             ArgumentNullException.ThrowIfNull(name);
             ArgumentNullException.ThrowIfNull(ownerType);
 
-            FromNameKey key = new FromNameKey(name, ownerType);
-
             while ((dp == null) && (ownerType != null))
             {
                 // Ensure static constructor of type has run
-                MS.Internal.WindowsBase.SecurityHelper.RunClassConstructor(ownerType);
+                SecurityHelper.RunClassConstructor(ownerType);
 
                 // Locate property
-                key.UpdateNameKey(ownerType);
-
+                FromNameKey key = new(name, ownerType);
                 lock (Synchronized)
                 {
-                    dp = (DependencyProperty)PropertyFromName[key];
+                    if (PropertyFromName.TryGetValue(key, out dp))
+                        return dp;
```

Comment:
@h3xds1nz I see that there is a behavioral change here if `dp` is null. In previous implementation, we were making sure that we have gone through all the objects of `FromNameKey` before returning a null. 
I am not sure if there is a case when dp stored in `PropertyFromName` can be null. What are your thoughts here? Can we ascertain that dp is never null? Also, if this is the case, I believe we can remove the null check from the while loop.

Reply:
@harshit7962 Since the value can never be `null` if the key is present, there's no behavioral change. You can use my mininal benchmark code for steps if you wish.

`PropertyFromName` is assigned under:
1) `AddOwner`, this assigns `this` as value (impossible to be `null`)
2) `RegisterCommon`, which creates a new instance via ctor (so it cannot be null also) and then assigns the variable to the dictionary, there's one operation inbetween (Seal) but that doesn't pass ref to the variable.

You wouldn't go through all the objects previously, once a value has been found under the key, you'd short-circuit too (hence why the `null` check for `dp` in while condition).

Remarks: `UpdateNameKey` is a full replica of a ctor besides the `name` that is immutable in the function.

I agree we can remove the check, it is rendered fully obsolete now. That's an oversight from my side.